### ISSUE TITLE
[Bug] ModelListGP posterior quietly ignores Log outcome transform

### ISSUE THREAD

**Issue by mrcslws** [pytorch/botorch#1519]
# 🐛 Bug

If you use a ModelListGP with Log outcome transforms on the contained models, the predictions are very bad. No error message is shown. If you step through the code, you find that `ModelListGP.posterior` ignores any outcome transforms that return a `TransformedPosterior` (e.g. Log, Power, Bilog).

This is not a recent regression, it has been quietly failing since at least botorch 0.6.1 (February), maybe forever.

## Why this fails

This code is broken 
https://github.com/pytorch/botorch/blob/92f0d1dfcca457278fa2a05b27bc9fffaac9b145/botorch/models/gpytorch.py#L616-L617

because TransformedPosterior doesn't have a `distribution` attribute, and this fact is hidden by
https://github.com/pytorch/botorch/blob/92f0d1dfcca457278fa2a05b27bc9fffaac9b145/botorch/models/gpytorch.py#L618
which was intended to handle the case where `outcome_transform` is missing, *not* `distribution`.

(This bug is independent of recent changes to this code, which renamed `mvn` to `distribution`.)

## To reproduce

```python
import botorch
import gpytorch
import torch

train_X = torch.tensor([[0., 0.],
                        [0.1, 0.1],
                        [0.5, 0.5]])
train_Y = torch.tensor([[0.1],
                        [0.5],
                        [1.0]])
test_X = torch.tensor([[0.3, 0.3]])

# Train and fit a single model.
model = botorch.models.SingleTaskGP(
    train_X, train_Y,
    # Use any transform that returns a TransformedPosterior
    outcome_transform=botorch.models.transforms.Log()
)

botorch.fit_gpytorch_mll(
    gpytorch.mlls.ExactMarginalLogLikelihood(model.likelihood, model)
)

# Access that model via a ModelListGP.
model_list = botorch.models.ModelListGP(model, model)

with torch.no_grad():
    result = model.posterior(test_X)
    list_result = model_list.posterior(test_X)

print(f"Model posterior: {result.mean, result.variance}")
print(f"ModelListGP posterior: {list_result.mean, list_result.variance}")
```

**Actual output**
```
Model posterior: (tensor([[1.3113]]), tensor([[4.3577]]))
ModelListGP posterior: (tensor([[-0.3602, -0.3602]]), tensor([[1.2625, 1.2625]]))
```

**Expected output**

Putting the model in a ModelListGP shouldn't change its predictions.

```
Model posterior: (tensor([[1.3113]]), tensor([[4.3577]]))
ModelListGP posterior: (tensor([[1.3113, 1.3113]]), tensor([[4.3577, 4.3577]]))
```

## System information

botorch: 0.7.3.dev22+g208470e7.d20221102 (latest)
gpytorch: 1.9.1.dev32+g23b068b5 (latest)
torch: 1.13.0
MacOS 13.0.1

**Comment by saitcakmak** [comment]
Thanks for flagging this and the detailed issue description! This is indeed a serious bug that was probably missed because we don't use these transforms much. 

I think a possible solution would be to catch when the posterior is `TransformedPosterior` and return a `PosteriorList` in those cases, rather than trying to can things into a single `GPyTorchPosterior`.

**Comment by esantorella** [comment]
Fixed by #1563 . Thanks again for reporting this!
